packages/core/src/utils.ts

```diff
@@ -328,30 +324,36 @@ export function isString(value: any): value is string {
 }
 
 export function toGuard<TContext, TEvent extends EventObject>(
-  condition?: Condition<TContext, TEvent>,
-  guardMap?: Record<string, ConditionPredicate<TContext, TEvent>>
-): Guard<TContext, TEvent> | undefined {
-  if (!condition) {
-    return undefined;
-  }
-
-  if (isString(condition)) {
+  guardConfig: GuardConfig<TContext, TEvent>,
+  guardMap?: Record<string, GuardPredicate<TContext, TEvent>>
+): GuardDefinition<TContext, TEvent> {
+  if (isString(guardConfig)) {
     return {
-      type: DEFAULT_GUARD_TYPE,
-      name: condition,
-      predicate: guardMap ? guardMap[condition] : undefined
+      type: guardConfig,
+      predicate: guardMap ? guardMap[guardConfig] : undefined,
+      params: { type: guardConfig }
```

Comment:
why the `type` is repeated in `params`? 🤔 shouldn't the `type` be enough here and `params` could just be empty?

Reply:
Because `type` is part of that object. There's no easy way to get the type here:

```js
guards: {
  someType: (context, event, meta) => {
    // ... how do you get the type??
  }
}
```

It's useful to know it and not beneficial to remove it.